src/Saml2/Provider.php

```diff
@@ -241,6 +260,11 @@ protected function sendMessage(SamlMessage $message, string $bindingType): HttpF
         $messageContext = new MessageContext();
         $messageContext->setMessage($message);
 
+        if (! empty(request()->filled('RelayState'))) {
```

Comment:
This would be better as:

```php
        if ($this->messageContext->getMessage() instanceof SamlMessage) {
            $messageContext->getMessage()->setRelayState($this->messageContext->getMessage()->getRelayState());
        }
```

So that it's referring to the same inbound messagecontext as the rest of the code.